tests/all/component_model/import.rs

```diff
@@ -235,28 +235,34 @@ fn attempt_to_leave_during_malloc() -> Result<()> {
         "bad trap: {}",
         trap,
     );
-    let trace = trap.trace().unwrap();
-    assert_eq!(trace.len(), 4);
-
-    // This was our entry point...
-    assert_eq!(trace[3].module_name(), Some("m"));
-    assert_eq!(trace[3].func_name(), Some("run"));
-
-    // ... which called an imported function which ends up being originally
-    // defined by the shim instance. The shim instance then does an indirect
-    // call through a table which goes to the `canon.lower`'d host function
-    assert_eq!(trace[2].module_name(), Some("host_shim"));
-    assert_eq!(trace[2].func_name(), Some("shim_ret_string"));
-
-    // ... and the lowered host function will call realloc to allocate space for
-    // the result
-    assert_eq!(trace[1].module_name(), Some("m"));
-    assert_eq!(trace[1].func_name(), Some("realloc"));
-
-    // ... but realloc calls the shim instance and tries to exit the
-    // component, triggering a dynamic trap
-    assert_eq!(trace[0].module_name(), Some("host_shim"));
-    assert_eq!(trace[0].func_name(), Some("shim_thunk"));
+
+    // TODO(#4535): we need to fold the Wasm<--->host trampoline functionality into
+    // component trampolines. Until then, the assertion about the stack trace in
+    // this test fails.
+    if false {
```

Comment:
To clarify, is this something you plan to take on or is this something I should plan to look into?

Reply:
I can look into the cranelift side of things but will probably need to ask you lots of questions about the component trampolines themselves.

Reply:
Ok sounds good to me. Given cranelift "intrinsics" for getting the fp/return-pc I think it'll be relatively easy to fold in the necessary support into the component trampolines.